python/python/lance/dataset.py

```diff
@@ -3898,6 +3898,7 @@ def write_dataset(
     use_legacy_format: Optional[bool] = None,
     enable_v2_manifest_paths: bool = False,
     enable_move_stable_row_ids: bool = False,
+    auto_cleanup_options: Optional[Dict[str, str]] = None,
```

Comment:
Can we make this a data class of some kind instead of `Dict[str, str]`?  This way users can more easily know what options are available and what they do?

Reply:
reasonable